### ISSUE TITLE
Weirdness with pfm_conditionals and pfm_exclude combination produces false exclusion

### ISSUE THREAD

**Issue by apizz** [erikberglund/ProfileCreator#109]
I've been working on updating the Accessibility manifest to add missing settings that can be managed, as well as add some menu structure that mirrors the organization found in the Accessibility Preference Pane in System Preferences.

In "Keyboard", there is an "Accessibility Keyboard" menu with some settings related to fading the accessibility keyboard panel.  When you enable the "Fade panel after inactivity" preference (`dwellHideUIEnabled`) you have the option to choose how many seconds of inactivity to fade this panel (`dwellHideUITimeout`) and also the percentage of how much to fade this panel by (`virtualKeyboardHideUITransparencyLevel`).

The desired manifest behavior is as follows:
- If `dwellHideUIEnabled` is set to `true`, automatically add both the `dwellHideUITimeout` and `virtualKeyboardHideUITransparencyLevel` preferences.
- If either `dwellHideUITimeout` or `virtualKeyboardHideUITransparencyLevel` is added w/o the other and the `dwellHideUIEnabled` preference also added, produce an exclusion message that these missing preferences need to be added.
- If `dwellHideUIEnabled` is NOT set to `true`, and either `dwellHideUITimeout` or `virtualKeyboardHideUITransparencyLevel` is added (or both), produce an exclusion message that `dwellHideUIEnabled` must be `true`.

The preferences are configured in the manifest as follows:

```
                  <dict>
			<key>pfm_description</key>
			<string>Preference fades the accessibility keyboard panel after configured dwellHideUITimeout seconds.</string>			
			<key>pfm_name</key>
			<string>dwellHideUIEnabled</string>
			<key>pfm_title</key>
			<string>Fade panel after inactivity</string>
			<key>pfm_type</key>
			<string>boolean</string>
		</dict>
		<dict>
			<key>pfm_conditionals</key>
			<array>
				<dict>
					<key>pfm_target_conditions</key>
					<array>
						<dict>
							<key>pfm_range_list</key>
							<array>
								<true/>
							</array>
							<key>pfm_target</key>
							<string>dwellHideUIEnabled</string>
						</dict>
					</array>
				</dict>
			</array>
			<key>pfm_default</key>
			<real>15</real>
			<key>pfm_description</key>
			<string>Fades the accessibility keyboard panel after X seconds of inactivity.</string>
			<key>pfm_exclude</key>
			<array>
				<dict>
					<key>pfm_target_conditions</key>
					<array>
						<dict>
							<key>pfm_present</key>
							<false/>
							<key>pfm_range_list</key>
							<array>
								<false/>
							</array>
							<key>pfm_target</key>
							<string>dwellHideUIEnabled</string>
						</dict>
						<dict>
							<key>pfm_present</key>
							<false/>
							<key>pfm_target</key>
							<string>virtualKeyboardHideUITransparencyLevel</string>
						</dict>
					</array>
				</dict>
			</array>
			<key>pfm_name</key>
			<string>dwellHideUITimeout</string>
			<key>pfm_title</key>
			<string>Fade Accessbility Keyboard panel after inactivity</string>
			<key>pfm_type</key>
			<string>float</string>
			<key>pfm_value_unit</key>
			<string>seconds</string>
		</dict>
		<dict>
			<key>pfm_conditionals</key>
			<array>
				<dict>
					<key>pfm_target_conditions</key>
					<array>
						<dict>
							<key>pfm_range_list</key>
							<array>
								<true/>
							</array>
							<key>pfm_target</key>
							<string>dwellHideUIEnabled</string>
						</dict>
					</array>
				</dict>
			</array>
			<key>pfm_description</key>
			<string>Preference sets the fade percentage of the Accessibility Keyboard after X seconds of inactivity.</string>
			<key>pfm_exclude</key>
			<array>
				<dict>
					<key>pfm_target_conditions</key>
					<array>
						<dict>
							<key>pfm_present</key>
							<false/>
							<key>pfm_range_list</key>
							<array>
								<false/>
							</array>
							<key>pfm_target</key>
							<string>dwellHideUIEnabled</string>
						</dict>
						<dict>
							<key>pfm_present</key>
							<false/>
							<key>pfm_target</key>
							<string>dwellHideUITimeout</string>
						</dict>
					</array>
				</dict>
			</array>
			<key>pfm_name</key>
			<string>virtualKeyboardHideUITransparencyLevel</string>
			<key>pfm_range_list</key>
			<array>
				<real>0.0</real>
				<real>0.05000000074505806</real>
				<real>0.10000000149011612</real>
				<real>0.15000000596046448</real>
				<real>0.20000000298023224</real>
				<real>0.25</real>
				<real>0.30000001192092896</real>
				<real>0.34999999403953552</real>
				<real>0.40000000596046448</real>
				<real>0.44999998807907104</real>
				<real>0.5</real>
				<real>0.55000001192092896</real>
				<real>0.60000002384185791</real>
				<real>0.64999997615814209</real>
				<real>0.69999998807907104</real>
				<real>0.75</real>
				<real>0.80000001192092896</real>
				<real>0.85000002384185791</real>
				<real>0.89999997615814209</real>
			</array>
			<key>pfm_range_list_titles</key>
			<array>
				<string>100%</string>
				<string>95%</string>
				<string>90%</string>
				<string>85%</string>
				<string>80%</string>
				<string>75%</string>
				<string>70%</string>
				<string>65%</string>
				<string>60%</string>
				<string>55%</string>
				<string>50%</string>
				<string>45%</string>
				<string>40%</string>
				<string>35%</string>
				<string>30%</string>
				<string>25%</string>
				<string>20%</string>
				<string>15%</string>
				<string>10%</string>
			</array>
			<key>pfm_title</key>
			<string>Fade by</string>
			<key>pfm_type</key>
			<string>float</string>
		</dict>
```

The strange behavior occurs when `dwellHideUIEnabled` is set to `true` and then adds both the `dwellHideUITimeout` and `virtualKeyboardHideUITransparencyLevel` preferences.  When added, despite the `virtualKeyboardHideUITransparencyLevel` preference being present the UI indicates that it's not.
![screen shot 2019-01-01 at 6 57 44 pm](https://user-images.githubusercontent.com/11789931/50577681-3f368200-0dfb-11e9-8552-6abcd633317b.png)

Interestingly, though both `dwellHideUITimeout` and `virtualKeyboardHideUITransparencyLevel` are configured require one another to be present, only one of these preferences produces an exclusion message.

Notably, when I add `dwellHideUITimeout` and `virtualKeyboardHideUITransparencyLevel` independently before adding and enabling `dwellHideUIEnabled`, the exclusion message is _not_ produced (see below).

**So, something with how the logic is processed in the `pfm_exclude` array is not working as expected when preferences are added as part of `pfm_conditionals`.**

![screen shot 2019-01-01 at 7 29 49 pm](https://user-images.githubusercontent.com/11789931/50577696-b5d37f80-0dfb-11e9-8ddf-e0a8f89bff2e.png)
![screen shot 2019-01-01 at 7 29 55 pm](https://user-images.githubusercontent.com/11789931/50577697-bc61f700-0dfb-11e9-87f5-81c5dfabf0e8.png)
![screen shot 2019-01-01 at 6 58 29 pm](https://user-images.githubusercontent.com/11789931/50577700-c2f06e80-0dfb-11e9-9ba4-2da3ffb1cb6b.png)
![screen shot 2019-01-01 at 6 58 39 pm](https://user-images.githubusercontent.com/11789931/50577701-c5eb5f00-0dfb-11e9-8b5a-60a7bd3ac18d.png)
